Automattic/woocommerce-services
# Make preloading optional for wc-admin prompt compatibility

Fixes #2154

This change makes it so the preloading of data is not required to open the label modal.

#1995 preloads data when the order page first loads (so additional requests are not needed), but requiring the preload breaks compatibility with the wc-admin order page prompt because it tries to open the label modal right after installing WCS.

With this WCS fix and the wc-admin fix( https://github.com/woocommerce/woocommerce-admin/pull/5025 ), the wc-admin order page prompt should be fully working again. It's a little challenging to test all of this together because the order page prompt will download the latest production WCS  from .org if none is installed. The fix here should only affect auto-loading on the order page, so we can still test it effectively by having WCS installed (**but not activated**) when the prompt is clicked.

To test:
1.  Install the wc-admin branch and follow the setup instructions to see the order page prompt to install WCS:
![image](https://user-images.githubusercontent.com/6209518/90933495-7680ea80-e3b4-11ea-8daa-d13c898551df.png)
2. Check out this branch (`fix/optional-preloading`) and build it. **WCS should be installed but deactivated**
3. If WCS is not activated, the prompt to install should still be visible.
4. Click "Create shipping label" button and wait
5. Since WCS is already installed, wc-admin should activate, load, and finally launch the label purchase modal.

While testing, we should also check that normal WCShip functionality is unaffected and that the label purchase modal loads OK and labels can be purchased.
